WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/ViewsAndVisitorsUseCase.kt

```diff
@@ -0,0 +1,277 @@
+                                "stats_age_in_days" to lastItemAge.toInt(),
+                                "is_jetpack_connected" to site.isJetpackConnected,
+                                "is_atomic" to site.isWPComAtomic,
+                                "action_source" to if (fetched) "remote" else "cached"
+                        )
+                )
+            }
+        }
+    }
+
+    override fun buildUiModel(
+        domainModel: VisitsAndViewsModel,
+        uiState: UiState
+    ): List<BlockListItem> {
+        val items = mutableListOf<BlockListItem>()
+        if (domainModel.dates.isNotEmpty()) {
+            items.add(buildTitle())
+
+            val dateFromProvider = selectedDateProvider.getSelectedDate(statsGranularity)
+            val visibleBarCount = uiState.visibleBarCount ?: domainModel.dates.size
+            val availableDates = domainModel.dates.map {
+                statsDateFormatter.parseStatsDate(
+                        statsGranularity,
+                        it.period
+                )
+            }
+            val selectedDate = dateFromProvider ?: availableDates.last()
+            val index = availableDates.indexOf(selectedDate)
+
+            selectedDateProvider.selectDate(
+                    selectedDate,
+                    availableDates.takeLast(visibleBarCount),
+                    statsGranularity
+            )
+            val selectedItem = domainModel.dates.getOrNull(index) ?: domainModel.dates.last()
+            val previousItem = domainModel.dates.getOrNull(domainModel.dates.indexOf(selectedItem) - 1)
+            items.add(
+                    viewsAndVisitorsMapper.buildTitle(
+                            selectedItem,
+                            previousItem,
+                            uiState.selectedPosition,
+                            isLast = selectedItem == domainModel.dates.last(),
+                            statsGranularity = statsGranularity
+                    )
+            )
+            items.addAll(
+                    viewsAndVisitorsMapper.buildChart(
+                            domainModel.dates,
+                            statsGranularity,
+                            this::onBarSelected,
+                            this::onBarChartDrawn,
+                            uiState.selectedPosition,
+                            selectedItem.period
+                    )
+            )
+            items.add(
+                    viewsAndVisitorsMapper.buildInformation()
+            )
+            items.add(
+                    viewsAndVisitorsMapper.buildChips(
+                            selectedItem,
+                            this::onColumnSelected,
+                            uiState.selectedPosition
+                    )
+            )
+        } else {
+            selectedDateProvider.onDateLoadingFailed(statsGranularity)
+            AppLog.e(T.STATS, "There is no data to be shown in the overview block")
```

Comment:
> ...overview block

We can name this views and visitors block.

Reply:
Good catch.  Yes, when we do graph part.  Also, there are items for no state and empty state on the board